spec/index.bs

```diff
@@ -101,10 +130,35 @@ partial interface WorkerNavigator {
 };
 </script>
 
-## GPUAdapter ## {#adapter}
+<script type=idl>
+[Exposed=Window]
+interface GPU {
+    // May reject with DOMException  // TODO: DOMException("OperationError")?
+    Promise<GPUAdapter> requestAdapter(optional GPURequestAdapterOptions options);
+};
+</script>
+
+## Adapters ## {#adapters}
+
+An <dfn dfn>adapter</dfn> represents an implementation of WebGPU on the system.
+Each adapter identifies both an instance of a hardware accelerator (e.g. GPU or CPU) and
+an instance of a software implementation of WebGPU on top of that accelerator.
+
+Note:
+An [=adapter=] may be a physical display adapter (GPU), but it could also be
+a software renderer.
+Different [=adapters=] could refer to different implementations on the
+same physical adapter (e.g. Vulkan and D3D12),
+or to different instances of the same physical configuration
+(e.g. if the GPU were disconnected and reconnected).
```

Comment:
why would it be a different adapter if it's just a GPU that re-connected?

Reply:
I was thinking the old adapter would be permanently lost, and we would return the new one if you ask again. This isn't strictly necessary (adapters don't have to be one-way the way devices do, according to ErrorHandling.md), but it keeps consistency. Imagine the following two cases:

- External GPU is unplugged and plugged back into the same port
- External GPU is unplugged and plugged back into a different port
- External GPU is unplugged and an identical one is plugged in

I see no reason that the application should want to be able to distinguish these, or should have to program against extra possibilities that behave the same way. So we either have to make these all appear as the same adapter (what happens once you have two?), or we have to make it always appear as a different adapter.

Reply:
I think the confusion comes from "Different" word here. I read it as: you have a list of things returned by a function, and this documentation explains why it has different things. But what you are trying to explain in this comment is why would a subsequent request for this list potentially return things that are different from the current request.

Reply:
I agree with @kainino0x that once an adapter becomes lost, it should stay permanently lost.  If the user re-plugs in an external GPU, WebGPU should create a brand new adapter object for it.

Reply:
Updated the text here; PTAL